### PR TITLE
test: Add storybook config fixture

### PR DESCRIPTION
**Pull request by askoufis**
Will be adding a few quality of life storybook config options alongside the upcoming storybook v7 release. This fixture will be used to test those config options.

Note: The fixture intentionally doesn't have any entrypoints since it should only be used for testing storybook.

### CODE REVIEW

**Review by jahredhope — APPROVED** [review]

File: fixtures/storybook-config/src/TestComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { Card, Text } from 'braid-design-system';
+import React, { ComponentProps } from 'react';
```
Comment (mrm007) [inline fixtures/storybook-config/src/TestComponent.tsx:2]:
```suggestion
import React, { type ComponentProps, type ReacNode } from 'react';
```

or 

```suggestion
import React from 'react';
```

File: fixtures/storybook-config/src/TestComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+type TestComponentProps = {
+  tone: ComponentProps<typeof Text>['tone'];
+  children: React.ReactNode;
```
Comment (mrm007) [inline fixtures/storybook-config/src/TestComponent.tsx:6]:
```suggestion
  tone: React.ComponentProps<typeof Text>['tone'];
  children: React.ReactNode;
```

or

```suggestion
  tone: ComponentProps<typeof Text>['tone'];
  children: ReactNode;
```

File: fixtures/storybook-config/src/TestComponent.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  decorators: [
+    (Story) => (
+      <StorybookDecorator>
+        <Story />
+      </StorybookDecorator>
+    ),
+  ],
+};
+
+export default meta;
```
Comment (mrm007) [inline fixtures/storybook-config/src/TestComponent.stories.tsx:18]:
```suggestion
export default {
  title: 'TestComponent',
  component: TestComponent,
  decorators: [
    (Story) => (
      <StorybookDecorator>
        <Story />
      </StorybookDecorator>
    ),
  ],
} satisfies Meta;
```

File: fixtures/storybook-config/src/TestComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+};
+
+export const TestComponent = ({ children, tone }: TestComponentProps) => (
+  <Card>
+    <Text tone={tone} data={{ ['automation-text']: true }}>
```
Comment (mrm007) [inline fixtures/storybook-config/src/TestComponent.tsx:11]:
```suggestion
    <Text tone={tone} data={{ 'automation-text': true }}>
```

**Review by mrm007 — APPROVED** [review]

**Pull request merged**